generator/postcode_points_builder.cpp

```diff
@@ -84,77 +86,53 @@ void GetPostcodes(std::string const & filename, storage::CountryId const & count
     if (std::find(countries.begin(), countries.end(), countryId) == countries.end())
       continue;
 
-    auto postcode = fields[fieldIndices.m_postcodeIndex];
-    if (!transformPostcode(postcode))
-      continue;
-
-    CHECK_EQUAL(valueMapping.size(), index, ());
+    keyValuePairs.emplace_back(search::NormalizeAndSimplifyString(fields[fieldIndices.m_postcodeIndex]),
+                               valueMapping.size());
     valueMapping.push_back(p);
-    keyValuePairs.emplace_back(search::NormalizeAndSimplifyString(postcode), Value(index));
-    ++index;
   }
 }
 
-template <typename Key, typename Value>
 void GetUKPostcodes(std::string const & filename, storage::CountryId const & countryId,
                     storage::CountryInfoGetter const & infoGetter,
-                    std::vector<m2::PointD> & valueMapping, std::vector<std::pair<Key, Value>> & keyValuePairs)
+                    std::vector<m2::PointD> & valueMapping, std::vector<ZipIndexValue> & keyValuePairs)
 {
-  // Original dataset uses UK National Grid UTM coordinates.
-  // It was converted to WGS84 by https://pypi.org/project/OSGridConverter/.
+  // * Follow these steps to prepare GB dataset: https://github.com/osm-search/gb-postcode-data
+  // * Direct from Nominatim: https://nominatim.org/data/gb_postcodes.csv.gz
+
   FieldIndices ukFieldIndices;
   ukFieldIndices.m_postcodeIndex = 0;
   ukFieldIndices.m_latIndex = 1;
   ukFieldIndices.m_longIndex = 2;
   ukFieldIndices.m_datasetCount = 3;
 
-  auto const transformUkPostcode = [](std::string & postcode) {
-    // UK postcodes formats are: aana naa, ana naa, an naa, ann naa, aan naa, aann naa.
-
-    // Do not index outer postcodes.
-    if (postcode.size() < 5)
-      return false;
-
-    // Space is skipped in dataset for |aana naa| and |aann naa| to make it fit 7 symbols in csv.
-    // Let's fix it here.
-    if (postcode.find(' ') == std::string::npos)
-      postcode.insert(static_cast<size_t>(postcode.size() - 3), " ");
-
-    return true;
-  };
-
-  GetPostcodes(filename, countryId, ukFieldIndices, ',' /* separator */, false /* hasHeader */,
-               transformUkPostcode, infoGetter, valueMapping, keyValuePairs);
+  GetPostcodes(filename, countryId, ukFieldIndices, "," /* separator */, true /* hasHeader */,
+               infoGetter, valueMapping, keyValuePairs);
 }
 
-template <typename Key, typename Value>
 void GetUSPostcodes(std::string const & filename, storage::CountryId const & countryId,
                     storage::CountryInfoGetter const & infoGetter,
-                    std::vector<m2::PointD> & valueMapping, std::vector<std::pair<Key, Value>> & keyValuePairs)
+                    std::vector<m2::PointD> & valueMapping, std::vector<ZipIndexValue> & keyValuePairs)
 {
-  // Zip;City;State;Latitude;Longitude;Timezone;Daylight savings time flag;geopoint
+  // * Get data source from here: https://simplemaps.com/data/us-zips
```

Comment:
Is all attribution present in data/copyright.html? @rtsisyk can you please double-check it?

Reply:
In theory we can use Nominatim and not simplemaps.com, but:
```
По поводу Nominatim dataset. Первый же entry по US оттуда: 00586,18.343697,-67.028434
Google Maps говорит что в этих координатах 00685
В simplemaps.com такого кода нет.
Google Maps такой код находит только в Африке.
```